android/quest/src/main/java/org/smartregister/fhircore/quest/ui/shared/components/ActionableButton.kt

```diff
@@ -64,78 +62,97 @@ fun ActionableButton(
+        ),
+      modifier =
+        modifier
+          .fillMaxWidth()
+          .padding(top = 0.dp, start = 12.dp, end = 12.dp)
+          .wrapContentHeight()
+          .testTag(ACTIONABLE_BUTTON_TEST_TAG)
     ) {
-      Spacer(modifier = modifier.weight(0.5f).fillMaxHeight())
-      Icon(
-        modifier = modifier.size(16.dp),
-        imageVector =
-          if (status == ServiceStatus.COMPLETED.name) Icons.Filled.Check else Icons.Filled.Add,
-        contentDescription = null,
-        tint =
-          when (status) {
-            ServiceStatus.COMPLETED.name -> SuccessColor.copy(alpha = 0.9f)
-            else -> buttonProperties.statusColor(computedValuesMap).copy(alpha = 0.9f)
-          }
-      )
-      Spacer(modifier = modifier.width(6.dp))
-      Text(
-        text = buttonProperties.text.toString(),
-        fontWeight = FontWeight.Medium,
-        color =
-          if (status == ServiceStatus.COMPLETED.name) DefaultColor.copy(0.9f)
-          else buttonProperties.statusColor(computedValuesMap).copy(alpha = 0.9f)
-      )
-      Spacer(modifier = Modifier.weight(0.5f).fillMaxHeight())
-      if (status == ServiceStatus.COMPLETED.name) {
+      Row(
+        verticalAlignment = Alignment.CenterVertically,
+        horizontalArrangement = Arrangement.Center,
+        modifier = modifier.wrapContentHeight().fillMaxWidth()
+      ) {
+        Spacer(modifier = modifier.weight(0.5f).wrapContentHeight())
         Icon(
-          imageVector = Icons.Filled.ArrowDropDown,
+          modifier = modifier.size(16.dp),
+          imageVector =
+            if (status == ServiceStatus.COMPLETED) Icons.Filled.Check else Icons.Filled.Add,
           contentDescription = null,
-          tint = DefaultColor.copy(alpha = 0.9f)
+          tint =
+            when (status) {
+              ServiceStatus.COMPLETED -> SuccessColor.copy(alpha = 0.9f)
+              else ->
+                buttonProperties.statusColor(resourceData.computedValuesMap).copy(alpha = 0.9f)
+            }
+        )
+        Spacer(modifier = modifier.width(6.dp))
+        Text(
+          text = buttonProperties.text?.interpolate(resourceData.computedValuesMap).toString(),
+          fontWeight = FontWeight.Medium,
+          color =
+            if (status == ServiceStatus.COMPLETED) DefaultColor.copy(0.9f)
+            else buttonProperties.statusColor(resourceData.computedValuesMap).copy(alpha = 0.9f)
         )
+        Spacer(modifier = modifier.weight(0.5f))
+        if (status == ServiceStatus.COMPLETED) {
+          Icon(
+            imageVector = Icons.Filled.ArrowDropDown,
+            contentDescription = null,
+            tint = DefaultColor.copy(alpha = 0.9f)
+          )
+        }
       }
     }
   }
 }
 
 @Composable
-fun ButtonProperties.statusColor(computedValuesMap: Map<String, Any>): Color = remember {
+fun ButtonProperties.statusColor(computedValuesMap: Map<String, Any>): Color {
   // Status color is determined from the service status
```

Comment:
can we move this comment to a docstring for the above fun/method?

Reply:
This has been updated